src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -655,7 +656,7 @@ private DefaultDeploymentTask createDefaultNewDeployment(Deployment deployment)
                     .kv("DeploymentType", deployment.getDeploymentType().toString())
                     .log("Invalid document for deployment");
             HashMap<String, String> statusDetails = new HashMap<>();
-            statusDetails.put("error", e.getMessage());
+            statusDetails.put(DEPLOYMENT_FAILURE_CAUSE_KEY, e.getMessage());
```

Comment:
why change this key? This may have unintended consequences.

Reply:
FSS only checks this and another key, it won't look at the key called "error": https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/status/FleetStatusService.java#L515-L516

But right we need to make sure nothing else is relying on the "error" key.